cli/cli.go

```diff
@@ -0,0 +1,163 @@
+// Package cli provides boilerplate code required to generate plugin executable.
+// At high level it performs following tasks
+// 1. Validate command arguments
+// 2. Read and unmarshal input
+// 3. Execute relevant plugin functions
+// 4. marshals output
+
+package cli
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"reflect"
+
+	"github.com/notaryproject/notation-plugin-framework-go/internal/slices"
+	"github.com/notaryproject/notation-plugin-framework-go/log"
+	"github.com/notaryproject/notation-plugin-framework-go/plugin"
+)
+
+type CLI struct {
```

Comment:
What is `CLI` struct and what does it do? Please add the struct doc.
We already have a struct called `CLIPlugin` in notation-go: https://github.com/notaryproject/notation-go/blob/966c6b7e42e463522cddc673bc753f67fe52de77/plugin/plugin.go#L72
Can we reuse it here? (I noticed that `CLIPlugin` is not copied from notation-go to this repo's `plugin/plugin.go`, why are we removing it?)

Reply:
This facilitates generation of plugin executable, see example plugin